commit e606d81d2d9596ab2b4fd0dc052eea0485b7e8c2
Merge: 12b7bcb43e6e b199ac6c4943
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Mon Oct 3 13:22:39 2016 -0700

    Merge branch 'ras-core-for-linus' of git://git.kernel.org/pub/scm/linux/kernel/git/tip/tip
    
    Pull RAS updates from Ingo Molnar:
     "The main changes were:
    
       - Lots of enhancements for AMD SMCA (Scalable MCA
         features/extensions) systems: extract, decode and print more
         hardware error information and add matching support on the
         injection/testing side as well. (Yazn Ghannam)
    
       - Various MCE handling improvements on modern Intel Xeons. (Tony
         Luck)
    
       - Plus misc fixes and enhancements"
    
    * 'ras-core-for-linus' of git://git.kernel.org/pub/scm/linux/kernel/git/tip/tip: (21 commits)
      x86/RAS/mce_amd_inj: Remove debugfs dir recursively on exit
      x86/RAS/mce_amd_inj: Fix signed wrap around when decrementing index 'i'
      x86/RAS/mce_amd_inj: Fix some W= warnings
      x86/MCE/AMD, EDAC: Handle reserved bank 4 on Fam17h properly
      x86/mce/AMD: Extract the error address on SMCA systems
      x86/mce, EDAC/mce_amd: Print MCA_SYND and MCA_IPID during MCE on SMCA systems
      x86/mce/AMD: Save MCA_IPID in MCE struct on SMCA systems
      x86/mce/AMD: Ensure the deferred error interrupt is of type APIC on SMCA systems
      x86/mce/AMD: Update sysfs bank names for SMCA systems
      x86/mce/AMD, EDAC/mce_amd: Define and use tables for known SMCA IP types
      EDAC/mce_amd: Use SMCA prefix for error descriptions arrays
      EDAC/mce_amd: Add missing SMCA error descriptions
      x86/mce/AMD: Read MSRs on the CPU allocating the threshold blocks
      x86/RAS: Add syndrome support to mce_amd_inj
      EDAC/mce_amd: Print syndrome register value on SMCA systems
      x86/mce: Add support for new MCA_SYND register
      x86/mce/AMD: Use msr_ops.misc() in allocate_threshold_blocks()
      x86/mce: Drop X86_FEATURE_MCE_RECOVERY and the related model string test
      x86/mce: Improve memcpy_mcsafe()
      x86/mce: Add PCI quirks to identify Xeons with machine check recovery
      ...

diff --cc include/linux/jump_label.h
index a534c7f15a61,595fb46213fc..a0547c571800
--- a/include/linux/jump_label.h
+++ b/include/linux/jump_label.h
@@@ -272,16 -273,9 +275,19 @@@ struct static_key_false 
  #define DEFINE_STATIC_KEY_FALSE(name)	\
  	struct static_key_false name = STATIC_KEY_FALSE_INIT
  
+ #define DECLARE_STATIC_KEY_FALSE(name)	\
+ 	extern struct static_key_false name
+ 
 +#define DEFINE_STATIC_KEY_ARRAY_TRUE(name, count)		\
 +	struct static_key_true name[count] = {			\
 +		[0 ... (count) - 1] = STATIC_KEY_TRUE_INIT,	\
 +	}
 +
 +#define DEFINE_STATIC_KEY_ARRAY_FALSE(name, count)		\
 +	struct static_key_false name[count] = {			\
 +		[0 ... (count) - 1] = STATIC_KEY_FALSE_INIT,	\
 +	}
 +
  extern bool ____wrong_branch_error(void);
  
  #define static_key_enabled(x)							\